test/embulk/input/zendesk/test_client.rb

```diff
@@ -482,6 +482,14 @@ def stub_response(status, headers = [], body_json = nil)
             stub_response(429, ["Retry-After: #{after}"])
             mock(client).sleep after.to_i
             assert_throw(:retry) do
+              client.tickets(false, &proc{})
+            end
+          end
+
+          test "429 guess/preview fail fast" do
+            after = "123"
+            stub_response(429, ["Retry-After: #{after}"])
+            assert_raise(DataError.new("Rate Limited. Waiting #{after} seconds to re-run")) do
```

Comment:
Can we assert that the method is executed only once, ie. no retry?

Reply:
Data Error means no retry https://github.com/treasure-data/embulk-input-zendesk/blob/3e749f3fc9af1889312a77f06f616f0a95886326/lib/embulk/input/zendesk/client.rb#L241 , isn't it ?

Reply:
It could be retried > gave up > and still `DataError`?

Reply:
I overlooked the test above this test, so it's good enough if it doesn't throw `:retry` (though some comments would be helpful).